src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -400,6 +402,28 @@ public virtual async Task<ConnectionCompleteParams> Connect(ConnectParams connec
             return completeParams;
         }
 
+        private async Task<ConnectionCompleteParams> TryOpenConnectionWithRetry(ConnectionInfo connectionInfo, ConnectParams connectionParams)
+        {
+            Boolean isRetriable = true;
+            int counter = 0;
+            ConnectionCompleteParams response = null;
+            while (isRetriable && counter <= MaxTries)
+            {
+                response = await TryOpenConnection(connectionInfo, connectionParams);
+                // If a serverless database is sleeping, it will return this error number and will need to be retried.
+                if (response?.ErrorNumber == 40613)
+                {
+                    counter++;
+                }
+                else
+                {
+                    // Every other response, we can stop.
+                    isRetriable = false;

```

Comment:
Just break out (or even directly return the response here) in this case, no need to have a separate var just for this